common/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/common/Codecs.scala

```diff
@@ -62,35 +64,40 @@ object Codecs {
       }
     }
 
-  implicit val inventoryDecoder: Decoder[InventoryItem] =
-    new Decoder[InventoryItem] {
-      def apply(c: HCursor): Decoder.Result[InventoryItem] =
+  implicit val inventoryDecoder: Decoder[ShreddedType] =
+    new Decoder[ShreddedType] {
+      def apply(c: HCursor): Decoder.Result[ShreddedType] =
         c.value.asObject match {
           case Some(obj) =>
             val map = obj.toMap
             for {
-              schema <- map.get("schema").liftTo[Decoder.Result](DecodingFailure("Property \"schema\" is not present", c.history))
-              schemaStr <- schema.asString.liftTo[Decoder.Result](DecodingFailure("Property \"schema\" is not a string", c.history))
-              shredPropertyStr <- map.get("type").liftTo[Decoder.Result](DecodingFailure("Property \"type\" is not present", c.history))
+              schema <- map.get("schema")
+                .liftTo[Decoder.Result](DecodingFailure("Property \"schema\" is not present", c.history))
+              schemaStr <- schema.asString
+                .liftTo[Decoder.Result](DecodingFailure("Property \"schema\" is not a string", c.history)).flatMap {
+                x => SchemaKey.fromUri(x).leftMap(e => DecodingFailure(e.code, Nil))
+              }
+              shredPropertyStr <- map
+                .get("type").liftTo[Decoder.Result](DecodingFailure("Property \"type\" is not present", c.history))
               shredProperty <- shredPropertyStr.as[ShredProperty]
-            } yield InventoryItem(shredProperty, schemaStr)
+            } yield ShreddedType(shredProperty, schemaStr)
           case None =>
             DecodingFailure(s"Cannot decode InventoryItem, ${c.value} is not an object", c.history).asLeft
```

Comment:
Not `InventoryItem` anymore.

Reply:
Again no idea, I think from other commits

Reply:
Yeah, I think we just forgot to fix it from 0.1.0